bundles/org.eclipse.equinox.launcher/src/org/eclipse/equinox/launcher/Main.java

```diff
@@ -1438,7 +1438,7 @@ public static void main(String[] args) {
 			// If the return code is 23, that means that Equinox requested a restart.
 			// In order to distinguish the request for a restart, do a System.exit(23)
 			// no matter of 'osgi.noShutdown' runtime property value.
-			if (!Boolean.getBoolean(PROP_NOSHUTDOWN) || result == 23)
+			if (!Boolean.getBoolean(PROP_NOSHUTDOWN) || result == 23 || result == 24)
```

Comment:
Ed, the comment has to be updated

Reply:
You think?  😱 😖

Reply:
Well, comment says something about 23, but then we have 23 and 24, with no idea why 24 is needed

Reply:
I was only joking.   If you read the linked issue you will know why 24 is needed.  In any case, note how the comment is actually wrong and or simply misleading already to suggest that it does System.exit(23) when the code actually does System.exit(result).  I'll fix it.